data/battle_scripts_1.s

```diff
@@ -20,6 +20,28 @@
 
 	.section script_data, "aw", %progbits
 
+BattleScript_EffectTidyUp::
```

Comment:
This _works_, but I recommend cleaning this up a bit. With some modifications, you should be able to get it working with something like:
```
attackcanceler
attackstring
ppreduce
trytidyup
goto BattleScript_EffectDragonDanceFromStatUp
```

If `trytidyup` can be triggered, then it should redirect to another battle script that plays the attackanimation and then handles the additional clearing effects. Finally after clearing the effects it checks if stats can be boosted and if they can then it jumps to Dragon Dance.

Reply:
When making the PR I tried to minimize the script length but I always ended up with a lot of bugs. Though I will give it an other go with the new hints.

Reply:
Not sure if this was the idea but I've made some changes that hopefully make the script more readable. I still had to add a pause because the attackstring disappeared almost immediately after showing up.